### ISSUE TITLE
Higher batch size is slower while training?

### ISSUE THREAD

**Issue by thepycoder** [ultralytics/ultralytics#1211]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Training

### Bug

I'm benchmarking YOLOv8 training on multiple GPUs for a blog post.

Using coco128, I'm observing slower average `epoch_time` for higher batch sizes. In contrast, I would expect the `epoch_time` to be lower the higher my batch size is set at. A single epoch should go through the whole dataset ( in this case coco128 with 128 sample images ). I have verified that the number of batches is correct in each case (2 batches of 64 batch_size, 4 for 32 and 8 for 16), so they each should be handling the same amount of images, yet my results seem to indicate that the opposite is true.

Any thoughts? I'm probably missing something really obvious here.
![batch_sizes](https://user-images.githubusercontent.com/11781950/222096304-5ad01b4e-5b51-4061-919c-52a0084454d3.png)

### Environment

- Ultralytics YOLOv8.0.42 🚀 Python-3.8.10 torch-1.13.1+cu117 CUDA:0 (NVIDIA GeForce RTX 3090, 24268MiB)
- nvidia/cuda:11.4.3-runtime-ubuntu20.04
![image](https://user-images.githubusercontent.com/11781950/222096699-e4088248-b908-4835-83c6-e92de53d42ee.png)

### Minimal Reproducible Example

```
from ultralytics import YOLO

args = dict(data="coco128.yaml", epochs=10, imgsz=640, task='detect', batch=16)
model_variant = "yolov8n"

# Load the YOLOv8 model
model = YOLO(f"{model_variant}.pt")  # load a pretrained model

# Train the model using our arguments from before
# If running remotely they may have been changed by ClearML
results = model.train(**args)
```

Change the batch size

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by AyushExel** [comment]
I think this can be due to the dataset size. 128 images are too less. Higher batch size might show improvements further into the epoch but for 128 images there are just 2-4 batches when using large batch size.
Could you try this test on the coco test set? I think that'll provide a better study and if this still happens, it can qualify as a bug.
